src/progress.rs

```diff
@@ -84,53 +73,69 @@ pub enum UpdateState<T> {
 }
 
 #[derive(Debug, Clone, Deserialize)]
-#[serde(rename_all = "camelCase")]
 pub struct SettingsUpdate {
     pub ranking_rules: UpdateState<Vec<RankingRule>>,
     pub distinct_attribute: UpdateState<String>,
-    pub identifier: UpdateState<String>,
+    pub primary_key: UpdateState<String>,
```

Comment:
What is the usage of `SettingsUpdate`? The user is not able to update the `primary_key` from the HTTP API, so I'm not sure this line was useful.

Reply:
I don't why it's available. This structure is sent by the server. I made everything available without thinking.

Reply:
Can you remove it then? I talk to the core-team and this structure does not represent what is the server sends, so this is not a concern of this SDK. There will never send a `primary_key` in the settings.

Reply:
Yeah, it seems you are right! I looked at the Meilisearch code and this seems to be dead code. A primary key cannot be updated with settings so it should not be possible for the server to notify the client that the key was updated. I will do a PR on meilisearch/meilisearch to fix this.

Reply:
Yes, it's useful for an internal purpose, this is not dead code. The core team told me you should avoid using the meilisearch-core structures since this is not what the MeiliSearch server will send. This is only internal.
It works for the moment, so we can let it as it is, but they do not recommend following exactly those files 🙂